producer-c/producer-cloudwatch-integ/KvsVideoOnlyStreamingSampleCloudwatch.cpp

```diff
@@ -0,0 +1,228 @@
+        clientConfiguration.region = region;
+        Aws::CloudWatch::CloudWatchClient cw(clientConfiguration);
+
+        // default storage size is 128MB. Use setDeviceInfoStorageSize after create to change storage size.
+        CHK_STATUS(createDefaultDeviceInfo(&pDeviceInfo));
+        // adjust members of pDeviceInfo here if needed
+        pDeviceInfo->clientInfo.loggerLogLevel = LOG_LEVEL_DEBUG;
+
+        CHK_STATUS(createRealtimeVideoStreamInfoProvider(streamName, DEFAULT_RETENTION_PERIOD, DEFAULT_BUFFER_DURATION, &pStreamInfo));
+        adjustStreamInfoToCanaryType(pStreamInfo, canaryType);
+        // adjust members of pStreamInfo here if needed
+        pStreamInfo->streamCaps.nalAdaptationFlags = NAL_ADAPTATION_FLAG_NONE;
+
+        CHK_STATUS(createDefaultCallbacksProviderWithAwsCredentials(accessKey,
+                                                                    secretKey,
+                                                                    sessionToken,
+                                                                    MAX_UINT64,
+                                                                    region,
+                                                                    cacertPath,
+                                                                    NULL,
+                                                                    NULL,
+                                                                    &pClientCallbacks));
+
+        if((retStatus = addFileLoggerPlatformCallbacksProvider(pClientCallbacks,
+                                                               CANARY_FILE_LOGGING_BUFFER_SIZE,
+                                                               CANARY_MAX_NUMBER_OF_LOG_FILES,
+                                                               (PCHAR) FILE_LOGGER_LOG_FILE_DIRECTORY_PATH,
+                                                               TRUE) != STATUS_SUCCESS)) {
+            printf("File logging enable option failed with 0x%08x error code\n", retStatus);
+        }
+        CHK_STATUS(createCanaryStreamCallbacks(&cw, streamName, &pStreamCallbacks, pCanaryStreamCallbacks));
+        CHK_STATUS(addStreamCallbacks(pClientCallbacks, pStreamCallbacks));
+        CHK_STATUS(createKinesisVideoClient(pDeviceInfo, pClientCallbacks, &clientHandle));
+        CHK_STATUS(createKinesisVideoStreamSync(clientHandle, pStreamInfo, &streamHandle));
+
+        // setup dummy frame
+        frame.size = CANARY_METADATA_SIZE + fragmentSizeInByte / DEFAULT_FPS_VALUE;
+        frame.frameData = (PBYTE) MEMALLOC(frame.size);
+        frame.version = FRAME_CURRENT_VERSION;
+        frame.trackId = DEFAULT_VIDEO_TRACK_ID;
+        frame.duration = 0;
+        frame.decodingTs = defaultGetTime(); // current time
+        frame.presentationTs = frame.decodingTs;
+
+        while (ATOMIC_LOAD_BOOL(&sigCaptureInterrupt) != TRUE) {
+            if (frameIndex < 0) {
```

Comment:
We might want to introduce a small variability in the frame size to mimmic the real stream. For example, the allocation above is OK. However, for every frame, we could set the actual size of the frame to be size - some smaller random value - say 10% of the size

Reply:
Good idea! As discussed, I will tackle it in another PR